crates/ruff/src/rules/flynt/rules/static_join_to_fstring.rs

```diff
@@ -27,15 +28,48 @@ impl AlwaysAutofixableViolation for StaticJoinToFString {
 }
 
 fn is_static_length(elts: &[Expr]) -> bool {
-    elts.iter().all(|e| !matches!(e, Expr::Starred(_)))
+    elts.iter().all(|e| !e.is_starred_expr())
 }
 
 fn build_fstring(joiner: &str, joinees: &[Expr]) -> Option<Expr> {
+    // If all elements are string constants, join them into a single string.
+    if joinees.iter().all(|expr| {
+        matches!(
+            expr,
+            Expr::Constant(ast::ExprConstant {
+                value: Constant::Str(_),
+                ..
+            })
+        )
+    }) {
+        let node = ast::ExprConstant {
+            value: Constant::Str(
+                joinees
+                    .iter()
+                    .filter_map(|expr| {
+                        if let Expr::Constant(ast::ExprConstant {
+                            value: Constant::Str(string),
+                            ..
+                        }) = expr
+                        {
+                            Some(string.as_str())
+                        } else {
+                            None
+                        }
+                    })
+                    .join(joiner),
+            ),
+            range: TextRange::default(),
+            kind: None,
+        };
+        return Some(node.into());
+    }
```

Comment:
This is a little more verbose, and it returns two passes, but it avoids allocating two vectors in every case.

Reply:
@charliermarsh Why do we want to avoid allocations instead of 2 passes in this case?

Reply:
My intuition is that the allocations would be more expensive than iterating over the vector a second time.

Reply:
Gotcha, thanks - just trying to get better :)

Reply:
Of course, I appreciate it — me too :)

I’m not an expert but I generally try to consider allocations expensive. In this case, we’re already iterating over this vec, so iterating over it again should be cheap, compared to the allocations at least.